billy-portugal-ebean/src/main/java/com/premiumminds/billy/portugal/persistence/dao/ebean/AbstractDAOPTGenericInvoiceImpl.java

```diff
@@ -30,14 +32,14 @@
     @SuppressWarnings("unchecked")
     @Override
     public TInterface findByNumber(UID uidBusiness, String number) {
-        /*QJPAPTGenericInvoiceEntity invoice = QJPAPTGenericInvoiceEntity.jPAPTGenericInvoiceEntity;
-        
-        return (TInterface) this.checkEntity(
-                this.createQuery()
-                        .from(invoice).where(this.toDSL(invoice.business, QJPAPTBusinessEntity.class).uid
-                                .eq(uidBusiness.toString()).and(invoice.number.eq(number)))
-                        .singleResult(invoice),
-                PTGenericInvoiceEntity.class);*/
-        return null;
+        JPAGenericInvoiceEntity invoice = this.queryInvoice(uidBusiness.toString(), number).findOne();
+        if (!(invoice instanceof PTGenericInvoiceEntity)) {
+            return null;
```

Comment:
The previous code queried PTGenericInvoiceEntities, and then checked if the invoice obtained was a PTGenericInvoiceEntity... if not it would throw a RuntimeException. In practice, that should never happen.
Due to the Ebean inheritance limitations, the new code has to query GenericInvoiceEntities (the superclass). So, after querying, the new code needs to check manually if the invoice is of the right class. In practice, it will always be the right class because the only concrete subclass is the PTGenericInvoiceEntity.

I will also throw a RuntimeException in this case, since it's something that should not happen.